mdn/developer-portal
# 135: Support autoformatting and linting with Therapist

This changeset contains a number of commits towards standardising the formatting and linting of the devportal project. (It wasn't bad before, just good to get this in during a quietish patch of development).

Closes #135 

## Summarised changes: 
* full-codebase passes made with `black`, `isort`, `flake8`, `eslint` and `prettier` to bring everything into line
* add [`therapist`](https://github.com/rehandalal/therapist) as a framework for running the above tooling (either via a pre-commit hook or directly - both paths are available)

+7,255 -3,637 looks huge, but the vast majority of the diff is standardising formatting

## To test (if you want to):

* pull this branch
* review the new section in the README about installing and using `therapist`
* do what the README says :o)
* confirm a full sweep works and detects no problems
* try committing a minor change that breaks formatting (eg: add some returns in a Python file and save it without a code-editor's black plugin fixing it) - confirm that therapist's pre-commit hook will block the commit
